shell/platform/windows/window_win32.cc

```diff
@@ -371,8 +371,11 @@ WindowWin32::HandleMessage(UINT const message,
       // Check if this key produces a character. If so, the key press should
       // be sent with the character produced at WM_CHAR. Store the produced
       // keycode (it's not accessible from WM_CHAR) to be used in WM_CHAR.
+      // As an exception with control or windows key pressed WM_CHAR may not
+      // come even though the event produced a character (i.e. CTRL + number).
```

Comment:
```suggestion
      //
      // Messages with Control or Win modifiers down are never considered as character
      // messages. This allows key combinations such as "CTRL + Digit" to properly
      // produce key down events even though `MapVirtualKey` returns a valid character.
      // See https://github.com/flutter/flutter/issues/85587.
```

Reply:
Also might be better to force `character` to be 0 when `GetKeyState(VK_CONTROL) == 0 && GetKeyState(VK_LWIN) == 0 && GetKeyState(VK_RWIN) == 0` for clarity.